### PR TITLE
generate sdk and bundle based on workspace path

### PR DESCRIPTION
**Pull request by stefandarius**
## Type of change

-   [x] 🐛 Bug Fix

## Description

SDK generation and bundling must be performed in the backend directory instead of the project root to provide the right context for language-specific tools such as the go parser.
We can achieve this by passing an optional path to the backend folder in the AST generator interface

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

File: src/generateSdk/sdkGenerator/TsSdkGenerator.ts [hunk 1]
```diff
@@ -292,7 +292,10 @@ class SdkGenerator implements SdkGeneratorInterface {
 
                 // @ts-expect-error A refactor need to be performed here to avoid this error
                 const methodView: MethodViewType = {
-                    name: methodDefinition.name,
+                    // methods start with lowercase
+                    name:
+                        methodDefinition.name.charAt(0).toLowerCase() +
+                        methodDefinition.name.slice(1),
```
Comment (costinsin) [inline src/generateSdk/sdkGenerator/TsSdkGenerator.ts:298]: Why is this change needed? What if i want to have functions that start with upper case? It's a ts convention to have lowercase named methods, but we should not impose it.

File: src/generateSdk/astGeneratorHandler.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ interface AstGeneratorPlugin {
 export async function generateAst(
     input: AstGeneratorInput,
     plugins: string[] | undefined,
+    backendPath?: string,
```
Comment (costinsin) [inline src/generateSdk/astGeneratorHandler.ts:30]: I would name it `rootPath` everywhere.
```suggestion
    rootPath?: string,
```

File: src/generateSdk/generateSdkApi.ts [hunk 3]
```diff
@@ -43,6 +43,7 @@ export async function sdkGeneratorApiHandler(
         const astGeneratorOutput = await generateAst(
             input,
             projectConfiguration.plugins?.astGenerator,
+            projectConfiguration.workspace?.backend,
```
Comment (costinsin) [inline src/generateSdk/generateSdkApi.ts:46]:
```suggestion
            /* rootPath= */ projectConfiguration.workspace?.backend,
```

File: src/generateSdk/astGeneratorHandler.ts (hunk 2 above)
Comment (costinsin) [reply src/generateSdk/astGeneratorHandler.ts:30]: Also, what if we include this path inside `AstGeneratorInput`? this way we could have this nice `AstGeneratorInput -> AstGeneratorOutput`, instead of `(AstGeneratorInput, path) -> AstGeneratorOutput`

**Review by vladiulianbogdan — APPROVED** [review]

File: src/generateSdk/sdkGenerator/TsSdkGenerator.ts (hunk 1 above)
Comment (stefandarius) [reply src/generateSdk/sdkGenerator/TsSdkGenerator.ts:298]: This is needed because all methods in Go will start with an uppercase letter.
The client code for our templates in ts/js expects methods starting with lowercase. We need this for a full-stack template with a go server and a ts client to work.

File: src/generateSdk/astGeneratorHandler.ts (hunk 2 above)
Comment (stefandarius) [reply src/generateSdk/astGeneratorHandler.ts:30]: i would find it a bit confusing to name it root since we also have the notion of the root of the mono repo project. i personally think of that as root, event though in the context of AST generation the "root of the action" is the backend folder.
i will try to see if I can add the value as a field in the `AstGeneratorInput` class

File: src/generateSdk/sdkGenerator/TsSdkGenerator.ts (hunk 1 above)
Comment (stefandarius) [reply src/generateSdk/sdkGenerator/TsSdkGenerator.ts:298]: @costinsin i added the root property in the AstGeneratorInput class. much cleaner now

**Review by costinsin — APPROVED** [review]
